daemon/status.go

```diff
@@ -45,33 +43,16 @@ func (d *Daemon) getK8sStatus() *models.K8sStatus {
 
 	}
 
-	var (
-		k8sStatus *models.K8sStatus
-		status    *k8sTypes.ComponentStatus
-		err       error
-	)
-	if status, err = k8s.Client().CoreV1().ComponentStatuses().Get("controller-manager", metav1.GetOptions{}); err != nil {
+	version, err := k8s.Client().Discovery().ServerVersion()
```

Comment:
Why not `k8s.GetServerVersion()`?

Reply:
I want to include the platform string as well.

Reply:
But be aware version.Major, version.Minor can be empty and `GetServerVersion()` is parsing it correctly by trying it first with the gitversion. I'm looking at the place where this will be. I think it's fine giving the fact that you are printing both major, minor and gitversion.

Reply:
The other reason is that we should avoid declaring cilium dead just because we can't parse the k8s version.